### ISSUE TITLE
[feature request] highlight relevant capture group

### ISSUE THREAD

**Issue by keith-hall** [SublimeText/PackageDev#92]
when writing syntax definitions, one quite often uses multiple capture groups. It can get quite tricky to find/remember where each group starts and which number it is.
I therefore propose that we add functionality similiar to how the syntax test highlighting helper works, and when the caret is on a `captures: n:` line in a `sublime-syntax` file, that the relevant group be highlighted in the corresponding `match:` line

```yaml
  - match: ((a)?(b)cde(fgh))(ij)
    captures:
      1: where is capture group 1 again?
```

**Comment by keith-hall** [comment]
here is the code I am using atm, far from PR-ready quality, but it works:

```python
def find_scope_within_region(view, region, scope_selector):
    for match in view.find_by_selector(scope_selector):
        if region.contains(match):
            yield match

def find_scope_before_region(view, region, scope_selector):
    prev_match = None
    for match in view.find_by_selector(scope_selector):
        if match.begin() < region.begin():
            prev_match = match
        else:
            break
    return prev_match

class HighlightCaptureGroupsViewEventListener(sublime_plugin.ViewEventListener):
    def on_selection_modified_async(self):
        if self.view.match_selector(self.view.sel()[0].begin(), 'source.yaml.sublime-syntax meta.expect-captures'):
            # get the capture group number
            group_num_region = next(find_scope_within_region(self.view, self.view.line(self.view.sel()[0].begin()), 'constant.numeric'), None)
            if group_num_region:
                group_num = int(self.view.substr(group_num_region))
                # find the region above the captures where a regex is expected
                check_region = find_scope_before_region(self.view, group_num_region, 'meta.expect-regexp')
                # ignore comments
                regex_pattern_regions = find_scope_within_region(self.view, check_region, 'source.regexp.oniguruma - comment')
                for region in regex_pattern_regions:
                    pos = region.begin()
                    while pos < region.end():
                        char = self.view.substr(pos)
                        if char == '\\': # skip escape chars
                            pos += 1
                        elif char == '(':
                            if self.view.substr(pos + 1) != '?': # if it is a capture group
                                group_num -= 1
                                if group_num == 0:
                                    pos_start = pos
                        elif char == ')':
                            if group_num == 0:
                                pos_end = pos + 1
                                region = sublime.Region(pos_start, pos_end)
                                self.view.add_regions('current_capture_group', [region], 'text', '', sublime.DRAW_NO_FILL)
                                return
                            elif group_num < 0:
                                group_num += 1 # nested groups
                        pos += 1
```

**Comment by FichteFoll** [comment]
Implemented in #102.
